### DISCUSSION TITLE
Sound of midi internal feature

### DISCUSSION THREAD

**Discussion by jjhesk** [4ian/GDevelop#3721] [Enhancement ideas (and their technical discussions)]
Lets build the midi support player in the gdevelop: https://github.com/cifkao/html-midi-player

**Comment by blurymind** [comment]
could use tone.js

renjs did something similar
https://renjs.net/examples/midi.html

https://github.com/lunafromthemoon/RenJS-V2/commit/e29ba4cd45b8ce6593498298f80eccad12367b77

**Reply by jjhesk** [reply]
Looking at online for the resources. There are alot of npm in the npmjs. Another browser ready module like: https://github.com/jazz-soft/jazz-midi
It is written in node-gyp and C++

**Reply by jjhesk** [reply]
the painful thing is that which sound font to best operate with the mid file or it allow user making custom sound font to play for their specific mid file?

**Reply by jjhesk** [reply]
tone.js have pretty good sound font.. but also it should allow user to add or replace for their custom soundfonts.

**Comment by jjhesk** [comment]
I am not expert of midi. Maybe someone can draft the first version for the look. We can make improvements afterward.
